Mergin/plugin.py

```diff
@@ -324,6 +331,38 @@ def configure_db_sync(self):
         if not wizard.exec_():
             return
 
+    def switch_sources(self):
+        project_path = QgsProject.instance().homePath()
+        if not project_path:
+            iface.messageBar().pushMessage("Mergin", "Project is not saved, please save project first", Qgis.Warning)
+            return
+
+        if not check_mergin_subdirs(project_path):
+            iface.messageBar().pushMessage(
+                "Mergin", "Current project is not a Mergin project. Please open a Mergin project first.", Qgis.Warning
+            )
+            return
+
+        mp = MerginProject(project_path)
+        try:
+            project_name = mp.metadata["name"]
+        except InvalidProject as e:
+            iface.messageBar().pushMessage(
+                "Mergin", "Current project is not a Mergin project. Please open a Mergin project first.", Qgis.Warning
+            )
+            return
+
+        dialog = ProjectUsePostgreConfigWizard(self.iface)
+        if not dialog.exec():
+            return
+
+        if check_mergin_subdirs(dialog.new_project_parent_folder()):
+            iface.messageBar().pushMessage(
+                "Mergin",
+                "The updated project should not be saved within Mergin directory. Please move it elsewhere.",
+                Qgis.Warning,
+            )
```

Comment:
I think this is a bit too late to say this :-) we should check that in the wizard dialog's code after picking output filename, and ask for a different output filename immediately